api/v1beta1/grafana_types.go

```diff
@@ -78,6 +78,13 @@ type GrafanaSpec struct {
 	External *External `json:"external,omitempty"`
 	// Preferences holds the Grafana Preferences settings
 	Preferences *GrafanaPreferences `json:"preferences,omitempty"`
+	// OperatorOptions holds settings to customize Grafana CR reconciliation by operator
+	OperatorOptions *GrafanaOperatorOptions `json:"operatorOptions,omitempty"`
+}
+
+type GrafanaOperatorOptions struct {
+	// DisableDefaultAdminSecret prevents operator from creating default admin-credentials secret
+	DisableDefaultAdminSecret bool `json:"disableDefaultAdminSecret,omitempty"`
```

Comment:
This week, we've discussed this change during maintainer's meeting. We're happy to see this functionality implemented, the only change that we'd like to see is moving `DisableDefaultAdminSecret` to the top level config (`GrafanaSpec`). - Strictly speaking, any configuration passed through CR is considered operator's options, and we don't have features that we'd like to group together under an extra section yet (once we do, we'll think about appropriate name).
For the rest, everything seems to work fine. Once you move the field, adjust references to it in the test and in `AdminSecretReconciler`, and regenerate manifests, we should be good to go.